src/js/ctrl/signals/defaults.ts

```diff
@@ -1,87 +1,98 @@
-'use strict';
-var dl = require('datalib'),
-    ns = require('../../util/ns'),
-    signals = {};
+// TODO(jzong): this probably should be moved to store/factory
 
-var SELECTED = ns('selected'),
-    MODE = ns('mode'),
-    ANCHOR = ns('anchor'),
-    DELTA = ns('delta'),
-    CURSOR = 'cursor',  // Special vega signal, don't namespace.
-    CELL = ns('cell'),
-    MOUSE = ns('mouse');
+import {Map, Record, RecordOf} from 'immutable';
+import {Signal as VegaSignal} from 'vega-typings/types';
+import ns from '../../util/ns';
 
-signals[SELECTED] = {
-  name: SELECTED,
-  init: {mark: {}},
-  streams: [
-    {type: 'mousedown[eventItem().mark && eventItem().mark.name &&' +
-        'eventItem().mark.name !== ' + dl.str(CELL) + ']',
-      expr: 'eventItem()'},
-    {type: 'mousedown[!eventItem().mark]', expr: '{mark: {}}'}
-  ],
-  _idx: 0
-};
-
-signals[MODE] = {
-  name: MODE,
-  init: 'handles',
-  _idx: 1
-};
+const dl = require('datalib');
 
-signals[DELTA] = {
-  name: DELTA,
-  init: 0,
-  streams: [
-    {type: '[mousedown, window:mouseup] > window:mousemove',
-      expr: '{x: eventX() - lyra_anchor.x, y: eventY() - lyra_anchor.y}'}
-  ],
-  _idx: 2
-};
+const SELECTED = ns('selected');
+const MODE = ns('mode');
+const ANCHOR = ns('anchor');
+const DELTA = ns('delta');
+const CURSOR = 'cursor';  // Special vega signal, don't namespace.
+const CELL = ns('cell');
+const MOUSE = ns('mouse');
 
-signals[ANCHOR] = {
-  name: ANCHOR,
-  init: 0,
-  streams: [
-    {type: 'mousedown',
-      expr: '{x: eventX(), y: eventY(), target: eventItem()}'},
-    {type: '[mousedown, window:mouseup] > window:mousemove',
-      expr: '{x: eventX(), y: eventY(), target: lyra_anchor.target}'}
-  ],
-  _idx: 3
-};
+export type LyraSignal = {
+  _idx: number
+} & VegaSignal;
+export const Signal = Record<LyraSignal>({
+  _idx: null,
+  name: null
+});
```

Comment:
See above comment. Because the definitions here aren't to be embedded into the store, we don't need to wrap it in a `Record` and can work with the `LyraSignal` directly.